src/ewatercycle/plugins/marrmot/model.py

```diff
@@ -77,29 +48,29 @@ class MarrmotM01(AbstractModel[MarrmotForcing]):
         <https://github.com/eWaterCycle/ewatercycle>`_
     """
 
-    model_name = "m_01_collie1_1p_1s"
-    """Name of model in Matlab code."""
-    available_versions = tuple(_version_images.keys())
-    """Versions for which ewatercycle grpc4bmi docker images are available."""
+    forcing: MarrmotForcing
+    bmi_image = ContainerImage("ewatercycle/marrmot-grpc4bmi:2020.11")
+
+    _model_name: str = PrivateAttr("m_01_collie1_1p_1s")
+    _parameters: List[float] = PrivateAttr([1000.0])
+    _store_ini: List[float] = PrivateAttr([900.0])
+    _solver: Solver = PrivateAttr(Solver())
+    _model_start_time: str | None = PrivateAttr()
+    _model_end_time: str | None = PrivateAttr()
 
-    def __init__(self, version: str, forcing: MarrmotForcing):  # noqa: D107
-        super().__init__(version, forcing=forcing)
-        self._parameters = [1000.0]
-        self.store_ini = [900.0]
-        self.solver = Solver()
-        self._check_forcing(forcing)
+    _forcing_filepath: str = PrivateAttr()
+    _forcing_start_time: datetime.datetime = PrivateAttr()
+    _forcing_end_time: datetime.datetime = PrivateAttr()
```

Comment:
I wonder if it makes sense to store all these as attributes separately on the model, or whether we could store them all in a single config object (can be a simple `dict`). Ideally we could use the models' native configuration files for that. So on creation you'd parse the template config file, update it in memory as necessary, then in `setup` save a copy with the latest changes and return that one.

Reply:
I am not sure if it matters too much if they're separate private attributes or items of a private attr dictionary.

Up to now I've mostly focused on getting everything moved over to the new API and working again. Seeing as these are all private attributes (i.e. not user accessible) we can change the internal workings (e.g., by rearranging them in a dict) without breaking things.